tools/Custom/ListCmdlet.cs

```diff
@@ -56,26 +52,47 @@ public partial class ListCmdlet : global::System.Management.Automation.PSCmdlet
         /// </summary>
         internal int totalFetchedItems = 0;
 
+        /// <summary>
+        /// Total number of items to be fetched.
+        /// </summary>
+        internal int limit = default;
+
         /// <summary>
         /// Initializes paging values.
         /// </summary>
         /// <param name="invocationInfo">A reference to <see cref="System.Management.Automation.InvocationInfo"/> object.</param>
-        /// <param name="PageSize">A reference to page size parameter.</param>
-        public void InitializePaging(ref global::System.Management.Automation.InvocationInfo invocationInfo, ref int PageSize)
+        /// <param name="top">A reference to top parameter.</param>
+        public void InitializePaging(ref global::System.Management.Automation.InvocationInfo invocationInfo, ref int top)
         {
-            if (PageSize > MaxPageSize)
+            if (invocationInfo.BoundParameters.ContainsKey("PageSize") && (PageSize > MaxPageSize || PageSize == default))
             {
-                invocationInfo.BoundParameters["All"] = true;
-                originalPageSize = PageSize;
-                requiredPages = PageSize / DefaultPageSize;
-                overflowItemsCount = PageSize % DefaultPageSize;
-                invocationInfo.BoundParameters["PageSize"] = DefaultPageSize;
-                PageSize = DefaultPageSize;
+                ThrowTerminatingError(
+                new ErrorRecord(
+                    new ArgumentException($"Invalid page size specified `{PageSize}`. {nameof(PageSize)} must be between 1 and {MaxPageSize} inclusive."),
+                    Guid.NewGuid().ToString(),
+                    ErrorCategory.InvalidArgument,
+                    null));
+            }
+
+            // Move `-Top` parameter to `limit` parameter.
+            if (invocationInfo.BoundParameters.ContainsKey("Top"))
+            {
+                limit = top;
+            }
+
+            // Explicitly set `-PageSize` as our $top parameter.
+            invocationInfo.BoundParameters["Top"] = invocationInfo.BoundParameters.ContainsKey("PageSize") ? PageSize : DefaultPageSize;
```

Comment:
why do we need to do this?

Reply:
We need it in order to construct a URL with a `$top` query parameter in the generated cmdlets. `PageSize` is a just a runtime parameter and is never serialized. Top/Limit on the other hand is a query parameter that's defined in the OpenAPI doc, and it will be serialized as `$top` query parameter.
``` cs
await this.Client.UsersUserListUser(this.InvocationInformation.BoundParameters.ContainsKey("Top") ? Top : default(int?), this.InvocationInformation.BoundParameters.ContainsKey("Skip") ? Skip : default(int?)...
 ```